### PR TITLE
[Release] 4.0.0

### LINKED ISSUE #2555: Test against WP 5.9
**Is your enhancement related to a problem? Please describe.**
Once WordPress 5.9 is released, we'll want to test ElasticPress to see if any incompatibility issues arise.

**Describe the solution you'd like**
- [ ] test ElasticPress on WP 5.9
- [ ] open issues for any incompatibilities noted in testing
- [ ] resolve issues identified in testing
- [ ] bump "tested up to" version
- [ ] if code changes needed due to incompatibilities, ship a plugin release, otherwise use "Plugin asset/readme update" action to update "tested up to" version on .org repo

**Designs**
n/a

**Describe alternatives you've considered**
none

**Additional context**
Related: #1708, #2124, #2250.

### PR DESCRIPTION
**Pull request by felipeelia**
This PR contains all the changes outlined in the release process (#2623)

Closes #2555 (see 1022510d9e0ba05e853ecbf3b04cc8b64e72be9a)

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 === ElasticPress ===
 Contributors: 10up, tlovett1, vhauri, tott, oscarssanchez, cmmarslender
 Tags:         performance, slow, search, elasticsearch, fuzzy, facet, aggregation, searching, autosuggest, suggest, elastic, advanced search, woocommerce, related posts, woocommerce
-Tested up to: 5.8
-Stable tag:   3.6.6
+Tested up to: 5.9.1
+Stable tag:   4.0.0
```
Comment (jeffpaul) [inline readme.txt:5]: Is there a reason this isn't bumped to `4.0.0-beta.1` as well?

File: CREDITS.md [hunk 2]
```diff
@@ -160,8 +160,17 @@ Thank you to all the people who have already contributed to this repository via
...
+[Stefan Fiedler (@Shrimpstronaut)](https://github.com/Shrimpstronaut),
+[Theo Savage (@tcrsavage)](https://github.com/tcrsavage),
+[Parham Ghaffarian (@ParhamG)](https://github.com/ParhamG),
+[Volodymyr Kolesnykov (@sjinks)](https://github.com/sjinks),
+[Siddharth Thevaril (@Sidsector9)](https://github.com/Sidsector9),
+[Mehidi Hassan (@mehidi258)](https://github.com/mehidi258),
+[Sathiya Venkatesan (@sathyapulse)](https://github.com/sathyapulse),
+[Dian Fishekqi (@dianfishekqi)](https://github.com/dianfishekqi),
+[@beazuadmin](https://github.com/beazuadmin),
 and
```
Comment (jeffpaul) [inline CREDITS.md:172]: I generally skip the `and` here as that ends up continually having to be shuffled around as new folks get added

File: readme.txt (hunk 1 above)
Comment (felipeelia) [reply readme.txt:5]: We didn't release the beta on WordPress.org, that is why we are going directly to 4.0.0.

Comment (jeffpaul) [reply readme.txt:5]: @felipeelia should other references then get changed to `4.0.0`?

Comment (felipeelia) [reply readme.txt:5]: Yup @jeffpaul, everything should be 4.0.0. Did I miss it somewhere?

Comment (jeffpaul) [reply readme.txt:5]: Yes, I'll make those updates once I finish with the changelog.md review

Comment (jeffpaul) [reply readme.txt:5]: Oops, totally misread the red/green changes, all clear on version numbers.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,83 @@ All notable changes to this project will be documented in this file, per [the Ke
 
 ## [Unreleased]
 
+## [4.0.0] - 2022-03-XX
+
+**ElasticPress 4.0 contains some important changes. Make sure to read these highlights before upgrading:**
+- This version requires a full reindex.
+- It introduces a new search algorithm that may change the order of results displayed on your site.
```
Comment (brandwaffle) [inline CHANGELOG.md:11]: "It introduces a new search algorithm that may change the search results displayed on your site." - theoretically more results could be matched in 4.0, not just the order, so let's make this a bit less specific.

Comment (felipeelia) [reply CHANGELOG.md]: Fixed in dfccf79, @brandwaffle .

**Pull request merged**